pkg/registry/apis/folders/folder_storage.go

```diff
@@ -150,7 +150,15 @@ func (s *folderStorage) DeleteCollection(ctx context.Context, deleteValidation r
 }
 
 func (s *folderStorage) setDefaultFolderPermissions(ctx context.Context, orgID int64, user identity.Requester, uid, parentUID string) error {
-	var permissions []accesscontrol.SetResourcePermissionCommand
+	isNested := parentUID != ""
+	if isNested {
+		return nil
+	}
+
+	permissions := []accesscontrol.SetResourcePermissionCommand{
```

Comment:
Should we check the `kubernetesDashboards` feature toggle ?
Like what's done here?
https://github.com/grafana/grafana/blob/e414e790f7410d22b1c337be7d4b7c2473ffd039/pkg/services/dashboards/service/dashboard_service.go#L1188